### PR TITLE
make sure no resource has the same id for capability type

### PR DESCRIPTION
**Pull request by jhsinger-klotho**
• Does any part of it require special attention?
• Does it relate to or fix any issue? closes #635

For now this only really will ever catch persist since no other resource type has multiple kinds, but checking them anyways in case they ever do.

### Standard checks

- **Unit tests**: Any special considerations? added
- **Docs**: Do we need to update any docs, internal or public? no
- **Backwards compatibility**: Will this break existing apps? If so, what would be the extra work required to keep them working? it shouldnt

### CODE REVIEW

File: pkg/validation/validation.go [hunk 1]
```diff
@@ -178,3 +197,16 @@ func getResourceById(id string, resources []core.CloudResource) core.ResourceKey
 	}
 	return resource
 }
+
+func validateNoDuplicateIds[T core.CloudResource](result *core.CompilationResult) error {
+	unitIds := make(map[string]struct{})
+	units := core.GetResourcesOfType[T](result)
+	for _, unit := range units {
+		fmt.Println(unit)
```
Comment (gordon-klotho) [inline pkg/validation/validation.go:205]:
```suggestion
```

File: pkg/validation/validation.go [hunk 2]
```diff
@@ -178,3 +197,16 @@ func getResourceById(id string, resources []core.CloudResource) core.ResourceKey
...
+		if id, ok := unitIds[unit.Key().Name]; ok {
+			return fmt.Errorf("Multiple Persist objects with the same name, '%s'", id)
```
Comment (gordon-klotho) [inline pkg/validation/validation.go:207]: Won't `id` be a `struct{}{}`?

Comment (jhsinger-klotho) [reply pkg/validation/validation.go:207]: ah yeah, meant to swithc that and log name. will update

**Comment by jhsinger-klotho** [comment]
fyi, this works, i needed to rebase from main locally because the bug from a couple days ago was still affecting this finding the persist annotations to begin with

**Review by ewucc — APPROVED** [review]

**Pull request merged**